python/tvm/relay/frontend/onnx.py

```diff
@@ -2737,6 +2737,12 @@ def from_onnx(model, shape=None, dtype="float32", opset=None, freeze_params=Fals
                 warnings.warn(str(e))
     except ImportError:
         pass
+
+    # if no explicit input's shape came from user, then initialize shape as it is defined in onnx model
+    if shape is None:
+        shape = {}
+        for i in model.graph.input:
+            shape[i.name] = [dim.dim_value for dim in i.type.tensor_type.shape.dim]
```

Comment:
I think we need to edit this logic just a little bit. If we have an unspecified input shape in the onnx model, it will show up as a variable, I think we need to identify that case and convert it to relay.Any() here.

For instance, this version of [BERT](https://github.com/mlcommons/inference/tree/master/language/bert) has for one of it's input shapes `[input_ids_dynamic_axes_1,384]` I believe this would fail on those inputs.